ffi/derive/src/impl_visitor.rs

```diff
@@ -195,8 +195,15 @@ impl<'ast> ImplDescriptor<'ast> {
 }
 
 impl<'ast> FnDescriptor<'ast> {
-    fn from_impl_method(self_ty: &'ast syn::Path, node: &'ast syn::ImplItemMethod) -> Self {
-        let mut visitor = FnVisitor::new(self_ty);
+    pub fn from_impl_method(self_ty: &'ast syn::Path, node: &'ast syn::ImplItemMethod) -> Self {
+        let mut visitor = FnVisitor::new(Some(self_ty));
+
+        visitor.visit_impl_item_method(node);
+        FnDescriptor::from_visitor(visitor)
+    }
+
+    pub fn from_fn(node: &'ast syn::ImplItemMethod) -> Self {
```

Comment:
I believe that from_fn should accept `&'ast syn::FnItem`. 
And here you have to options:
1) Put conversation that you already have inside this function:
```rust
    pub fn from_fn(node: &'ast syn::FnItem) -> Self {
          #[allow(clippy::expect_used)]
          let impl_method =
              syn::parse(item.to_token_stream().into()).expect("Can't parse fn item");
          // ...
    }
```
2. Second option would be to extend `FnVisitor` by adding method `visit_item_fn` (it would be almost the same as `vist_impl_item_method`).